runtime-modules/bounty/src/lib.rs

```diff
@@ -0,0 +1,676 @@
+    pub creator_funding: Balance,
+}
+
+// Helper enum for the bounty management.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum BountyCreator<MemberId> {
+    // Bounty was created by a council.
+    Council,
+
+    // Bounty was created by a member.
+    Member(MemberId),
+}
+
+impl<MemberId> Default for BountyCreator<MemberId> {
+    fn default() -> Self {
+        BountyCreator::Council
+    }
+}
+
+/// Defines current bounty stage.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum BountyStage {
+    /// Bounty founding stage with starting block number.
+    Funding,
+
+    /// A bounty was canceled.
+    Canceled,
+
+    /// Funding and cherry can be withdrawn.
+    Withdrawal,
+
+    /// A bounty has gathered necessary funds and ready to accept work submissions.
+    WorkSubmission,
+}
+
+/// Defines current bounty state.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum BountyMilestone<BlockNumber> {
+    /// Bounty was created at given block number.
+    Created(BlockNumber),
+
+    /// A bounty was canceled.
+    Canceled,
+
+    /// A bounty funding was successful on the provided block.
+    /// The stage is set when the funding exceeded max funding amount.
+    MaxFundingReached(BlockNumber),
+}
+
+impl<BlockNumber: Default> Default for BountyMilestone<BlockNumber> {
+    fn default() -> Self {
+        BountyMilestone::Created(Default::default())
+    }
+}
+
+/// Alias type for the Bounty.
+pub type Bounty<T> = BountyRecord<
+    BalanceOf<T>,
+    <T as frame_system::Trait>::BlockNumber,
+    <T as common::Trait>::MemberId,
+>;
+
+/// Crowdfunded bounty record.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct BountyRecord<Balance, BlockNumber, MemberId> {
+    /// Bounty creation parameters.
+    pub creation_params: BountyParameters<Balance, BlockNumber, MemberId>,
+
+    /// Total funding balance reached so far.
+    /// Includes initial funding by a creator and other members funding.
+    pub total_funding: Balance,
+
+    /// Bounty current state. It represents fact known about the bounty, eg.:
+    /// it was canceled or max funding amount was reached.
+    pub state: BountyMilestone<BlockNumber>,
+}
+
+/// Balance alias for `balances` module.
+pub type BalanceOf<T> = <T as balances::Trait>::Balance;
+
+decl_storage! {
+    trait Store for Module<T: Trait> as Bounty {
+        /// Bounty storage
+        pub Bounties get(fn bounties) : map hasher(blake2_128_concat) T::BountyId => Bounty<T>;
+
+        /// Double map for bounty funding. It stores member funding for bounties.
+        pub Funding get(fn funding_by_bounty_by_member): double_map hasher(blake2_128_concat)
```

Comment:
Can we rename `Funding` to `BountyContributions`

Reply:
Done